auth/src/main/java/com/firebase/ui/auth/viewmodel/AuthViewModelBase.java

```diff
@@ -1,27 +1,26 @@
 package com.firebase.ui.auth.viewmodel;
 
 import android.app.Application;
-import android.app.PendingIntent;
 import android.arch.lifecycle.LiveData;
-import android.content.Intent;
+import android.arch.lifecycle.MutableLiveData;
 import android.support.annotation.RestrictTo;
 import android.support.annotation.VisibleForTesting;
 
 import com.firebase.ui.auth.data.model.FlowParameters;
+import com.firebase.ui.auth.data.model.Resource;
 import com.firebase.ui.auth.util.GoogleApiUtils;
 import com.google.android.gms.auth.api.credentials.CredentialsClient;
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.PhoneAuthProvider;
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-public class AuthViewModelBase extends ViewModelBase<FlowParameters> {
-
+public abstract class AuthViewModelBase<T> extends ViewModelBase<FlowParameters> {
     private CredentialsClient mCredentialsClient;
     private FirebaseAuth mAuth;
     private PhoneAuthProvider mPhoneAuth;
 
-    private SingleLiveEvent<PendingResolution> mPendingResolution = new SingleLiveEvent<>();
+    private MutableLiveData<Resource<T>> mOperation = new MutableLiveData<>();
```

Comment:
Can you quickly explain how you're mitigating the risk that comes with removing the `SingleLiveEvent` for the errors?

Reply:
You got it below, it's the `setUsed` thing. I'm totally open to going back to our `SingleLiveEvent`, but that felt like the greater of the two evils. Thoughts?